frontend/src/routes/Settings.tsx

```diff
@@ -17,18 +17,84 @@ function Settings() {
         enabled: !!nodeManager,
     })
 
+    function handleSave() {
+        let serializable: Record<string, any> = {};
+        for (let i = 0; i < localStorage.length; i++) {
+            const key = localStorage.key(i)
+            serializable[key!] = localStorage.getItem(key!);
+        }
+        console.log(serializable)
+
+        saveTemplateAsFile("mutiny_wallet_backup.json", serializable)
+    }
+
+    // https://stackoverflow.com/questions/19721439/download-json-object-as-a-file-from-browser
+    const saveTemplateAsFile = (filename: string, dataObjToWrite: Record<string, any>) => {
+        const blob = new Blob([JSON.stringify(dataObjToWrite)], { type: "text/json" });
+        const link = document.createElement("a");
+
+        link.download = filename;
+        link.href = window.URL.createObjectURL(blob);
+        link.dataset.downloadurl = ["text/json", link.download, link.href].join(":");
+
+        const evt = new MouseEvent("click", {
+            view: window,
+            bubbles: true,
+            cancelable: true,
+        });
+
+        link.dispatchEvent(evt);
+        link.remove()
+    };
+
+    function handleFileChoose(e: React.ChangeEvent) {
+        const fileReader = new FileReader();
+        const target = e.target as HTMLInputElement;
+        const file: File = (target.files as FileList)[0];
+        fileReader.readAsText(file, "UTF-8");
+        fileReader.onload = e => {
+            handleClearState();
```

Comment:
lets not clear state until we we absolutely have to, in case something goes wrong. This should happen after the console log I think